### PR TITLE
Visitor Can See Articles Grouped By Category

### PR DESCRIPTION
**Pull request by tochman**
PT: https://www.pivotaltracker.com/story/show/161218473

* Creates Category Model 
* Adds logic to filter by category to the articles controller#index action
* Creates association between Category and Article
* Add navigation between categories

### What I learned:
To sand my ground. Rupert wanted me to go out of scope, but I resisted.

### CODE REVIEW

File: features/visitor_can_see_articles_on_landing_page.feature [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
-  
+      | header        | subheader             | body                     | category    |
+      | Top title     | A breaking news item  | Today at craftacademy    | Programming |
+      | Other stories | Another breaking news | Tomorrow at craftacademy | Education   |
+
+
+  Scenario: View available articles on the landing page
     When I visit the site
+    And I click on "Programming"
+    Then stop
```
Comment (rupertlion) [inline features/visitor_can_see_articles_on_landing_page.feature]: We need to remove this - semaphore build not completing.

File: app/models/article.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Article < ApplicationRecord
+
+    belongs_to :category
     validates :header, presence: true
```
Comment (rupertlion) [inline app/models/article.rb:4]: We should create a chore to refactor these to validates_presence_of

**Review by rupertlion — CHANGES_REQUESTED** [review]
Looks good to me apart from the "stop" command to make the build work.

File: features/visitor_can_see_articles_on_landing_page.feature (hunk 1 above)
Comment (tochman) [reply features/visitor_can_see_articles_on_landing_page.feature]: Wow! I missed that. Sorry.

File: app/models/article.rb (hunk 2 above)
Comment (tochman) [reply app/models/article.rb:4]: In a refactoring chore. Not now IMO

**Review by rupertlion — APPROVED** [review]

File: app/models/article.rb (hunk 2 above)
Comment (rupertlion) [reply app/models/article.rb:4]: agreed - hence "We should create a chore to refactor these to validates_presence_of" :)

**Pull request merged**